pallets/multisig/src/lib.rs

```diff
@@ -888,68 +917,74 @@ impl<T: Trait> Module<T> {
             !Self::votes(&multisig_signer_proposal),
             Error::<T>::AlreadyVoted
         );
-        if let Some(proposal) = Self::proposals(&multisig_proposal) {
-            let mut proposal_details = Self::proposal_detail(&multisig_proposal);
-            proposal_details.approvals += 1u64;
-            let current_did = Context::current_identity::<Identity<T>>().unwrap_or_default();
-            match proposal_details.status {
-                ProposalStatus::Invalid => return Err(Error::<T>::ProposalMissing.into()),
-                ProposalStatus::Rejected => return Err(Error::<T>::ProposalAlreadyRejected.into()),
-                ProposalStatus::ExecutionSuccessful | ProposalStatus::ExecutionFailed => {}
-                ProposalStatus::ActiveOrExpired => {
-                    // Ensure proposal is not expired
-                    if let Some(expiry) = proposal_details.expiry {
-                        ensure!(
-                            expiry > <pallet_timestamp::Module<T>>::get(),
-                            Error::<T>::ProposalExpired
-                        );
-                    }
-                    Self::execute_proposal(
+        ensure!(
+            <Proposals<T>>::contains_key(&multisig_proposal),
+            Error::<T>::ProposalMissing
+        );
+
+        let mut proposal_details = Self::proposal_detail(&multisig_proposal);
+        proposal_details.approvals += 1u64;
+        let multisig_did = <MultiSigToIdentity<T>>::get(&multisig);
+        match proposal_details.status {
+            ProposalStatus::Invalid => return Err(Error::<T>::ProposalMissing.into()),
+            ProposalStatus::Rejected => return Err(Error::<T>::ProposalAlreadyRejected.into()),
+            ProposalStatus::ExecutionSuccessful | ProposalStatus::ExecutionFailed => {}
+            ProposalStatus::ActiveOrExpired => {
+                // Ensure proposal is not expired
+                if let Some(expiry) = proposal_details.expiry {
+                    ensure!(
+                        expiry > <pallet_timestamp::Module<T>>::get(),
+                        Error::<T>::ProposalExpired
+                    );
+                }
+                if proposal_details.approvals >= Self::ms_signs_required(&multisig) {
+                    let execution_at = system::Module::<T>::block_number() + One::one();
+                    let call = Call::<T>::execute_scheduled_proposal(
                         multisig.clone(),
                         proposal_id,
-                        proposal,
-                        &mut proposal_details,
-                        current_did,
-                    )?;
+                        multisig_did,
+                    )
+                    .into();
+
+                    // Scheduling will fail when it's already scheduled (had enough votes already).
+                    // We ignore the failure here.
+                    let _ = T::Scheduler::schedule_named(
```

Comment:
I think it is good to emit an event for the failure case.

Reply:
Neh, it will just be noise IMO.